File: pegasusX/apps/retailer-app-ios/retailerapp/Views/Dock/ShelfAlertModal.swift
import SwiftUI

struct ShelfAlertModal: View {
    let productId: String
    let currentStock: Int64
    let onDismiss: () -> Void

    var body: some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.red)
            
            Text("RESTOCK NEEDED")
                .font(.title2)
                .bold()
                .foregroundColor(Color(red: 0.6, green: 0.1, blue: 0.1)) // Dark red
            
            Text("Product: \(productId) has dropped to \(currentStock) items on the floor.")
                .font(.subheadline)
                .multilineTextAlignment(.center)
                .foregroundColor(Color(red: 0.5, green: 0.1, blue: 0.1))
            
            Button(action: onDismiss) {
                Text("ACKNOWLEDGE")
                    .font(.headline)
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .padding()
                    .background(Color.red)
                    .cornerRadius(8)
            }
            .padding(.top, 8)
        }
        .padding(24)
        .background(Color(red: 1.0, green: 0.9, blue: 0.9)) // Light red background
        .cornerRadius(16)
        .shadow(radius: 10)
        .padding(24)
    }
}

File: pegasusX/apps/retailer-app-android/app/src/main/java/com/pegasusx/retailer/ui/screens/dock/ShelfAlertModal.kt
package com.pegasusx.retailer.ui.screens.dock

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog

@Composable
fun ShelfAlertModal(
    visible: Boolean,
    productId: String,
    currentStock: Long,
    onDismiss: () -> Unit
) {
    if (!visible) return

    Dialog(onDismissRequest = onDismiss) {
        Card(
            shape = RoundedCornerShape(12.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFFFEE2E2)),
            elevation = CardDefaults.cardElevation(8.dp),
            modifier = Modifier.fillMaxWidth().padding(16.dp)
        ) {
            Column(
                modifier = Modifier.padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Icon(
                    imageVector = Icons.Default.Warning,
                    contentDescription = "Alert",
                    tint = Color(0xFFDC2626),
                    modifier = Modifier.size(48.dp)
                )
                Spacer(modifier = Modifier.height(16.dp))
                Text(
                    text = "RESTOCK NEEDED",
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color(0xFF991B1B)
                )
                Spacer(modifier = Modifier.height(8.dp))
                Text(
                    text = "Product: $productId has dropped to $currentStock items on the floor.",
                    fontSize = 14.sp,
                    color = Color(0xFF7F1D1D),
                    textAlign = androidx.compose.ui.text.style.TextAlign.Center
                )
                Spacer(modifier = Modifier.height(24.dp))
                Button(
                    onClick = onDismiss,
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFDC2626)),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text("ACKNOWLEDGE", color = Color.White, fontWeight = FontWeight.Bold)
                }
            }
        }
    }
}
